packages/extension-text-style/src/text-style.ts

```diff
@@ -64,15 +63,32 @@ export const TextStyle = Mark.create<TextStyleOptions>({
 
   addCommands() {
     return {
-      removeEmptyTextStyle: () => ({ state, commands }) => {
-        const attributes = getMarkAttributes(state, this.type)
-        const hasStyles = Object.entries(attributes).some(([, value]) => !!value)
+      removeEmptyTextStyle: () => ({ tr }) => {
 
-        if (hasStyles) {
-          return true
-        }
+        const { selection } = tr
 
-        return commands.unsetMark(this.name)
+        // Gather all of the nodes within the selection range.
+        // We would need to go through each node individually
+        // to check if it has any inline style attributes.
+        // Otherwise, calling commands.unsetMark(this.name)
+        // removes everything from all the nodes
+        // within the selection range.
+        tr.doc.nodesBetween(selection.from, selection.to, (node, pos) => {
+
+          // Check if it's a paragraph element, if so, skip this node as we apply
+          // the text style to inline text nodes only (span).
+          if (node.isTextblock) {
+            return true
+          }
+
+          // Check if the node has no inline style attributes.
+          if (!node.marks.some(mark => Object.values(mark.attrs).some(value => !!value))) {
```

Comment:
This should also check that the mark is the textStyle mark no?

We don't want to remove attributes on just any mark

Reply:
It applies to all nodes between those two positions. To get nodes that have the textStyle mark on them, you'd need to add some additional filtering like:

```js
node.marks.some(mark => mark.type.name === this.type) && !node.marks.some(mark => Object.values(mark.attrs...
```

Reply:
Yeah, that makes sense, I thought that within the selection most of them would be working correctly, I'd need to check with other nodes too just to make sure! Going back at it 🔧